src/services/GoalsService.ts

```
@@ -0,0 +1,516 @@
+          if (!goal.category) return 0;
+          return await this.calculateCategoryTime(goal.category, startOfDay, endOfDay);
+
+        case 'app_limit':
+          // Time spent on specific app
+          if (!goal.appName) return 0;
+          return await this.calculateAppTime(goal.appName, startOfDay, endOfDay);
+
+        case 'weekly_time':
+          // For weekly goals, calculate from start of week
+          const startOfWeek = this.getStartOfWeek(date);
+          return await this.calculateTotalActiveTime(startOfWeek, endOfDay);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)